tests/RandomTest.php

```diff
@@ -497,6 +497,43 @@ public function testNewInstance()
         $this->assertTrue(isset($a->persistence));
     }
 
+    public function testDuplicateSaveNew()
+    {
+        $this->setDb([
+            'rate' => [
+                ['dat' => '18/12/12', 'bid' => 3.4, 'ask' => 9.4],
+                ['dat' => '12/12/12', 'bid' => 8.3, 'ask' => 9.2],
+            ],
+        ]);
+
+        $db = new Persistence\Sql($this->db->connection);
+        $m = new Model_Rate($db);
+
+        $m->load(1)->duplicate()->save();
```

Comment:
this is weird with word `duplicate` as save will create a new record

we should rename it to `duplicateAndUnload` at least. Then this PR LGTM.

Reply:
The whole point of duplicating is to use the existing data for a new record. We are not unloading, only clearing the entityId.
What we can do is include the `save` already in the method.

Reply:
I see, data are retained, LGTM